roles/origin_file_modify/tasks/main.yml

```diff
@@ -0,0 +1,23 @@
+---
+# vim: set ft=ansible:
+#
+# This role modifies the origin file refspec in order to use
+# rpm-ostree upgrade directly
+#
+# Parameters:
+#  refspec (string) - refspec to add to booted origin file
+#
+- name: Fail if refspec is undefined
+  fail:
+    msg: "refspec is undefined"
+  when: refspec is undefined
+
+- name: Get current booted dir
+  command: ostree admin --print-current-dir
+  register: current_dir
+
+- name: Replace current refspec
+  replace:
+    dest: "{{ current_dir.stdout }}.origin"
+    regexp: '^(.*refspec=).*$'
+    replace: '\1{{ refspec }}'
```

Comment:
I figured this was possible, but I couldn't easily find it in any man pages or help...until some Google-fu lead me to this little snippet:

`rpm-ostree is now capable of rebasing to a local branch using the rebase :local-branch syntax`

http://www.projectatomic.org/blog/2017/06/rpm-ostree-v2017.6-released/

Sure, enough after creating the local branch, we can just rebase to that and then do the upgrade:

```
# rpm-ostree status
State: idle
Deployments:
● fedora-atomic:fedora-atomic/25/x86_64/docker-host
                Version: 25.148 (2017-06-25 21:07:13)
                 Commit: f136ff5370c0f77d9c1f4fd3ecb8e1815650d8aaf834e41a5e995a13494407e9
# ostree refs --create local-branch f136ff5370c0f77d9c1f4fd3ecb8e1815650d8aaf834e41a5e995a13494407e9
# rpm-ostree rebase :local-branch
Copying /etc changes: 22 modified, 0 removed, 57 added
Transaction complete; bootconfig swap: yes deployment count change: 1
Run "systemctl reboot" to start a reboot
# rpm-ostree status
State: idle
Deployments:
  local-branch
                Version: 25.148 (2017-06-25 21:07:13)
                 Commit: f136ff5370c0f77d9c1f4fd3ecb8e1815650d8aaf834e41a5e995a13494407e9

● fedora-atomic:fedora-atomic/25/x86_64/docker-host
                Version: 25.148 (2017-06-25 21:07:13)
                 Commit: f136ff5370c0f77d9c1f4fd3ecb8e1815650d8aaf834e41a5e995a13494407e9

... reboot system ...

# ostree commit -b local-branch --tree=ref=local-branch --add-metadata-string version=test1
f1cc261cffc29167bbbed04a3c83ce11c62ebe131dbdd89e11abbc08ad287a8b
# rpm-ostree upgrade
Copying /etc changes: 22 modified, 0 removed, 59 added
Transaction complete; bootconfig swap: no deployment count change: 0
Run "systemctl reboot" to start a reboot
# atomic host status
State: idle
Deployments:
  local-branch
                Version: test1 (2017-07-11 20:42:48)
                 Commit: f1cc261cffc29167bbbed04a3c83ce11c62ebe131dbdd89e11abbc08ad287a8b

● local-branch
                Version: 25.148 (2017-06-25 21:07:13)
                 Commit: f136ff5370c0f77d9c1f4fd3ecb8e1815650d8aaf834e41a5e995a13494407e9
```